src/MyInfoGovClient.class.ts

```diff
@@ -1,609 +1,331 @@
-   *          "occupation": {
-   *            "lastupdated": "2017-10-11",
-   *            "source": "2",
-   *            "classification": "C",
-   *            "value": "53201",
-   *            "desc": "HEALTHCARE ASSISTANT"
-   *          },
-   *          "employment": {
-   *            "lastupdated": "2017-10-11",
-   *            "source": "2",
-   *            "classification": "C",
-   *            "value": "ALPHA"
-   *          },
-   *          "workpassstatus": {
-   *            "lastupdated": "2018-03-02",
-   *            "source": "1",
-   *            "classification": "C",
-   *            "value": "Live"
-   *          },
-   *          "workpassexpirydate": {
-   *            "lastupdated": "2018-03-02",
-   *            "source": "1",
-   *            "classification": "C",
-   *            "value": "2018-12-31"
-   *          },
-   *          "householdincome": {
-   *            "high": "5999",
-   *            "low": "5000",
-   *            "lastupdated": "2017-10-24",
-   *            "source": "2",
-   *            "classification": "C"
-   *          },
-   *          "vehno": {
-   *            "lastupdated": "",
-   *            "source": "2",
-   *            "classification": "C",
-   *            "value": ""
-   *          }
-   *        }
+   * Constructs a redirect URL which the user can visit to initialise
+   * SingPass login and consent to providing the given MyInfo attributes.
+   * @param config Configuration object
+   * @param config.purpose Purpose of requesting the data, which will be
+   * shown to user
+   * @param config.requestedAttributes MyInfo attributes which the user must
+   * consent to provide
+   * @param config.relayState Optional state to be forwarded to the redirect endpoint
+   * via query parameters
+   * @param config.singpassEserviceId Optional alternative e-service ID.
+   * Defaults to the e-serviceId provided in the constructor.
+   * @param config.redirectEndpoint Optional alternative redirect endpoint.
+   * Defaults to the endpoint provided in the constructor.
+   * @returns The URL which the user should visit to log in to SingPass
+   * and consent to providing the given attributes.
    */
-  getPersonBasic ({
-    uinFin,
+  createRedirectURL({
+    purpose,
     requestedAttributes,
-    txnNo,
-    singpassEserviceId: seId,
-  }: IPersonBasicRequest): Promise<IPersonBasic> {
-    if (!requestedAttributes || requestedAttributes.length === 0) {
-      requestedAttributes = ALL_ATTRIBUTES
+    relayState,
+    singpassEserviceId,
+    redirectEndpoint,
+  }: IAuthRequest): string {
+    const queryParams = {
+      purpose,
+      attributes: requestedAttributes.join(),
+      state: relayState ?? '',
+      client_id: this.clientId,
+      redirect_uri: redirectEndpoint ?? this.redirectEndpoint,
+      sp_esvcId: singpassEserviceId ?? this.singpassEserviceId,
     }
+    return `${this.baseAPIUrl}${Endpoint.Authorise}?${qs.stringify(
+      queryParams,
+    )}`
```

Comment:
i think this would be more readable

```suggestion
    const queryParams = qs.stringify({
      purpose,
      attributes: requestedAttributes.join(),
      state: relayState ?? '',
      client_id: this.clientId,
      redirect_uri: redirectEndpoint ?? this.redirectEndpoint,
      sp_esvcId: singpassEserviceId ?? this.singpassEserviceId,
    })
    return `${this.baseAPIUrl}${Endpoint.Authorise}?${queryParams}`
```